src/REISE.jl

```diff
@@ -68,12 +69,18 @@ function run_scenario(;
     println("All scenario files loaded!")
     case = reise_data_mods(case; num_segments=num_segments)
     save_input_mat(case, storage, inputfolder, outputfolder)
-    model_kwargs = Dict(
+    # Create final model kwargs by merging defaults with user-specified
+    default_model_kwargs = Dict(
         "case" => case,
         "storage" => storage,
         "demand_flexibility" => demand_flexibility,
         "interval_length" => interval,
     )
+    if isnothing(model_kwargs)
+        model_kwargs = default_model_kwargs
+    else
+        model_kwargs = merge(default_model_kwargs, model_kwargs)
```

Comment:
What does `merge` do? I presume it should work similarly as `dict.update` in Python, right?

Reply:
Exactly, except since Julia is based around multiple dispatch we can call `merge` on either pairs of `AbstractDict` objects (of which `Dict` is a sub-class) or `NamedTuple` objects, or could define another `merge` method that merges other sorts of objects.

https://docs.julialang.org/en/v1/base/collections/#Base.merge